code/processes/housekeeping.q

```diff
@@ -74,13 +74,16 @@ kdbzip:{[FILE]
 \d .unix
 
 //-locates files with path, matching string and age
-find:{[path;match;age;agemin]
+find:{[path;match;age;agemin;checkfordirectory]
+	$[0=checkfordirectory;
+		[fileordir::"files";flag::"f"];
+		[fileordir::"directories";flag::"d"]];
 	$[0=agemin;
-	files:.[{.lg.o[`housekeeping;"Searching for: ",x,y];system "/usr/bin/find ", x," -maxdepth 1 -type f -name \"",y,"\" -mtime +",raze string z};(path;match;age);
+	matches:.[{.lg.o[`housekeeping;"Searching for ",fileordir,": ",x,y];system "/usr/bin/find ", x," -maxdepth 1 -type ",flag," -name \"",y,"\" -mtime +",raze string z};(path;match;age);
 	{.lg.e[`housekeeping;"Find function failed: ", x]; ()}];
-	 files:.[{.lg.o[`housekeeping;"Searching for: ",x,y];system "/usr/bin/find ", x," -maxdepth 1 -type f -name \"",y,"\" -mmin +",raze string z};(path;match;age);
+	 matches:.[{.lg.o[`housekeeping;"Searching for ",fileordir,": ",x,y];system "/usr/bin/find ", x," -maxdepth 1 -type ",flag," -name \"",y,"\" -mmin +",raze string z};(path;match;age);
```

Comment:
I know you didn't do it like this originally, but could you tidy this up? These two lines are basically the same, except for the mtime/mmin switch. So it would be better just to have this all written out once.